### PR TITLE
Check Skylake fixes

### PR DESCRIPTION
**Pull request by plbossart**
Our Intel colleagues have a set of cleanups that potentially affect SOF.
Let's run a round of reviews and CI

### CODE REVIEW

File: sound/hda/ext/hdac_ext_bus.c [hunk 1]
```diff
@@ -172,7 +172,6 @@ EXPORT_SYMBOL_GPL(snd_hdac_ext_bus_device_init);
 void snd_hdac_ext_bus_device_exit(struct hdac_device *hdev)
 {
 	snd_hdac_device_exit(hdev);
-	kfree(hdev);
 }
```
Comment (libinyang) [inline sound/hda/ext/hdac_ext_bus.c:175]: hdev may be allocated by kzalloc(), please refer: sound/soc/sof/intel/hda-codec.c

**Comment by libinyang** [comment]
Sorry, I don't mean the patchset is OK to me. Wrongly click the approve. Actually, the first patch is not right. 
The 2, 3, 5 patch is OK to me.

**Comment by plbossart** [comment]
> As compr is allocated by kzalloc() in the original code, there should be kfree() when the driver is removed. But currently I can't find kfree() of compr. Maybe the original code forgot to free the compr.
> 
> The patch is OK to me.

@libinyang What code are your referring to? It seems our SOF code is wrong then? What are you trying to say?

**Comment by libinyang** [comment]
@plbossart  Sorry, I was referring to the patch 2 "ASoC: compress: Fix memory leak from snd_soc_new_compress " in that comments. I'm OK with the patch 2.

> @libinyang What code are your referring to? It seems our SOF code is wrong then? What are you trying to say?

File: sound/hda/ext/hdac_ext_bus.c (hunk 1 above)
Comment (bardliao) [reply sound/hda/ext/hdac_ext_bus.c:175]: We should keep it consistent. I.e. we should use the same function to allocate hdec on both skylake and sof driver.

**Comment by plbossart** [comment]
> @plbossart Sorry, I was referring to the patch 2 "ASoC: compress: Fix memory leak from snd_soc_new_compress " in that comments. I'm OK with the patch 2.

please don't top-post in replies, it makes them hard to read.

> > @libinyang What code are your referring to? It seems our SOF code is wrong then? What are you trying to say?

I was trying to figure out what the flow was for the hdev kzalloc/free issue you commented on. You didn't explain if there is an SOF-specific way of allocating memory that's different from what the Skylake/hdac_ library expect.

**Comment by libinyang** [comment]
Please don't merge the first patch, or let make a additional patch for the first patch.

**Comment by plbossart** [comment]
> Please don't merge the first patch, or let make a additional patch for the first patch.

for the 3rd time, please explain your concerns. You did not provide a clear answer as to why this is problematic for SOF and not for SST w/ hdac.

**Comment by libinyang** [comment]
> for the 3rd time, please explain your concerns. You did not provide a clear answer as to why this is problematic for SOF and not for SST w/ hdac.

if you look into sound/soc/sof/intel/hda-codec.c: hda_codec_probe(), it will call:
hdev = zalloc(sizeof(*hdev), GFP_KERNEL); to allocate the hdev.
In this case, kfree(hdev) is necessary. Otherwise, the memory will be leaked.

So I suggested to make the change to hdev = devm_kzalloc() together in this patch.

**Comment by plbossart** [comment]
> if you look into sound/soc/sof/intel/hda-codec.c: hda_codec_probe(), it will call:
> hdev = zalloc(sizeof(*hdev), GFP_KERNEL); to allocate the hdev.
> In this case, kfree(hdev) is necessary. Otherwise, the memory will be leaked.
> 
> So I suggested to make the change to hdev = devm_kzalloc() together in this patch.

can you please submit a PR with your change and the removal of the the kfree (keep Amadeusz authorship)? As @bardliao said, if we use a common function and allocate the memory in different ways it's a recipe for disaster...

**Comment by libinyang** [comment]
> > if you look into sound/soc/sof/intel/hda-codec.c: hda_codec_probe(), it will call:
> > hdev = zalloc(sizeof(*hdev), GFP_KERNEL); to allocate the hdev.
> > In this case, kfree(hdev) is necessary. Otherwise, the memory will be leaked.
> > So I suggested to make the change to hdev = devm_kzalloc() together in this patch.
> 
> can you please submit a PR with your change and the removal of the the kfree (keep Amadeusz authorship)? As @bardliao said, if we use a common function and allocate the memory in different ways it's a recipe for disaster...

Sure, I will do it.

**Comment by libinyang** [comment]
@plbossart done
Please check: https://github.com/thesofproject/linux/pull/928

**Pull request closed without merging**